### ISSUE TITLE
Add PcdPkcs7CertBufferXdr to SecurityPkg that supports one or more PKCS7 certificates (Bugzilla Bug 891)

### ISSUE THREAD

**Issue by tianocore-issues** [tianocore/edk2#7267]
This issue was created automatically with [bugzilla2github](https://github.com/makubacki/bugzilla2github/tree/bz2gh_tianocore)
# Bugzilla Bug 891
Date: 2018-03-12T14:30:53+00:00
From: @mdkinney
To:   @mdkinney
CC:   

Duplicates:   #922
Depends on:   #890
Last updated: 2018-08-09T16:39:37+00:00

**Comment by tianocore-issues** [comment]
## Comment 3677
Date: 2018-03-12 14:30:53 +0000
From: @mdkinney

- Industry Specification: ---
- Release Observed: EDK II Master
- Releases to Fix: EDK II Master
- Target OS: ---
- Bugzilla Assignee(s): @mdkinney

The SecurityPkg currently provide the PCD PcdPkcs7CertBuffer to specify
a single PKCS7 certificate.

There are capsule update use cases that require support for more than
one PKCS7 certificate.

A new PCD PcdPkcs7CertBufferXdr should be added to the SecurityPkg that
supports one more PKCS7 certificates encoded using the Variable-Length
Opaque Data format of RFC 4506 External Data Representation Standard (XDR).

The default value for this new PCD should be {0} (empty).

The enhancements to the BaseTools BinToPcd tool can be used to generate
a PCD using this format from multiple certificate files.

    https://bugzilla.tianocore.org/show_bug.cgi?id=890

Both PcdPkcs7CertBuffer and PcdPkcs7CertBufferXdr should be used to
authenticate a capsule.  The capsule fails authentication of none of
the certificates in PcdPkcs7CertBuffer and PcdPkcs7CertBufferXdr pass.

Both PcdPkcs7CertBuffer and PcdPkcs7CertBufferXdr should be evaluated
to check for the use of the test key.  If the test key is found in either
PCD, then the warning messages for the use of a test key must be 
presented.

**Comment by tianocore-issues** [comment]
## Comment 3680
Date: 2018-03-12 16:19:45 +0000
From: @mdkinney

Assign to owner

**Comment by tianocore-issues** [comment]
## Comment 3681
Date: 2018-03-12 16:20:58 +0000
From: @mdkinney

Patch email

https://lists.01.org/pipermail/edk2-devel/2018-March/022632.html

Branch for review

https://github.com/mdkinney/edk2/tree/Bug_890_891_BinToPcdMultipleInputFiles

**Comment by tianocore-issues** [comment]
## Comment 3740
Date: 2018-03-21 18:56:10 +0000
From: @mdkinney

It would be better to add a new PCD that supports multiple PKCS7
certificates to a package that is focused on capsules instead of
the SecurityPkg

**Comment by tianocore-issues** [comment]
## Comment 4563
Date: 2018-08-09 16:39:37 +0000
From: @mdkinney

A PCD that supports one or more PKCS7 certificates in XDR format was added
to the FmpDevicePkg as part of:

    https://bugzilla.tianocore.org/show_bug.cgi?id=922

The details of this PCD are as follows:

PcdsFixedAtBuild, PcdsPatchableInModule, PcdsDynamic, PcdsDynamicEx]
  ## One or more PKCS7 certificates used to verify a firmware device capsule
  #  update image.  Encoded using the Variable-Length Opaque Data format of RFC
  #  4506 External Data Representation Standard (XDR).  The default value is
  #  empty with 0 certificates.
  # @Prompt One or more XDR encoded PKCS7 certificates used to verify firmware device capsule update images.
  gFmpDevicePkgTokenSpaceGuid.PcdFmpDevicePkcs7CertBufferXdr|{0x0}|VOID*|0x4000000E

*** This bug has been marked as a duplicate of bug 922 ***
